### PR TITLE
Orca bigdl estimator eval add validation_metrics None error msg

### PR DESCRIPTION
**Pull request by cyita**

### CODE REVIEW

**Comment by cyita** [comment]
related issue: #3437

**Comment by leonardozcm** [comment]
hi,  in case users pass `metrics=['accuracy']`, which is not support by bigdl, will raise another error. Maybe you should also check the type of validation_metrics? And also add it to `estimator.fit` would make it perfect.

File: pyzoo/zoo/orca/learn/bigdl/estimator.py [hunk 1]
```diff
@@ -180,6 +180,9 @@ def predict(self, data, batch_size=4, feature_cols="features", sample_preprocess
     def evaluate(self, data, batch_size=32, feature_cols=None, label_cols=None,
                  validation_metrics=None):
         assert data is not None, "validation data shouldn't be None"
+        assert validation_metrics is not None, "validation_metrics shouldn't be None, please " \
```
Comment (jason-dai) [inline pyzoo/zoo/orca/learn/bigdl/estimator.py:183]: then why do we make `validation_metrics` default to None?

File: pyzoo/zoo/orca/learn/bigdl/estimator.py [hunk 2]
```diff
@@ -180,7 +186,12 @@ def predict(self, data, batch_size=4, feature_cols="features", sample_preprocess
     def evaluate(self, data, batch_size=32, feature_cols=None, label_cols=None,
                  validation_metrics=None):
         assert data is not None, "validation data shouldn't be None"
+        if validation_metrics is None:
+            validation_metrics = [Accuracy()]
```
Comment (jason-dai) [inline pyzoo/zoo/orca/learn/bigdl/estimator.py:190]: Why different behaviors for `fit` vs. `evaluate`?

Comment (cyita) [reply pyzoo/zoo/orca/learn/bigdl/estimator.py:190]: Is merging upstream master and metrics will default to be Accuracy when init the estimator.

File: pyzoo/zoo/orca/learn/pytorch/estimator.py [hunk 3]
```diff
@@ -223,6 +225,8 @@ def __init__(self, model, loss, optimizer, metrics=None, model_dir=None, bigdl_t
         else:
             raise ValueError("Only PyTorch optimizer and orca optimizer are supported")
         from zoo.orca.learn.metrics import Metrics
+        if metrics is None:
+            metrics = Accuracy()
```
Comment (yangw1234) [inline pyzoo/zoo/orca/learn/pytorch/estimator.py:229]: changing to default loss?

**Comment by cyita** [comment]
Jenkins passed http://10.239.47.210:18888/view/ZOO-PR-PIPELINE/job/ZOO-PR-Validation/5041/ 
http://10.239.47.210:18888/job/ZOO-PR-Validation-Other/5044/console

**Pull request merged**